doc/GettingStarted.md

```diff
@@ -0,0 +1,225 @@
+# Getting Started
+
+On this page you will learn about [TSL](#TSL) and its Generator, what it is inteded for and how to utilize it.
+
+
+# Introduction
+
+### **What is TSL and what is it used for?**
+
+TSL is a C++ Header-only Library which abstracts elemental operations on 
+hardware specific instructions sets, espacial SIMD / vector operations.
+
+### **Why the indirection through a generator? Why no plain header only library?**
+
+While developing TSL we found, that a large potion is redundant boiler plate code. 
+When looking into the generated files, this should be clear.
+So to avoid writing the same boiler plate over and over again, we implemented the generator.
+Also the generator automatically detects the CPU flags set on your system, 
```

Comment:
+ While this seems to be unnecessary at first glance, through (i) modern compilers also support hardware detection for auto-vectorization and (ii) non-compliant code parts can be disabled using preprocessor macros, we want to support (i) explicit vectorization and reduce the code size and compile time and increase the readability of the to be shipped library (ii).